src/meshio/netgen/_netgen.py

```diff
@@ -0,0 +1,350 @@
+
+        with gzip.open(filename, "rt") as f:
+            return read_buffer(f)
+
+    with open_file(filename, "r") as f:
+        return read_buffer(f)
+
+
+def _read_cells(f, netgen_cell_type, cells, cells_index):
+    if netgen_cell_type == "pointelements":
+        dim = 0
+        nump = 1
+        pi0 = 0
+        i_index = 1
+    if netgen_cell_type.startswith("edgesegments"):
+        dim = 1
+        nump = 2
+        pi0 = 2
+        i_index = 0
+    if netgen_cell_type == "surfaceelements":
+        dim = 2
+        i_index = 1
+    if netgen_cell_type == "volumeelements":
+        dim = 3
+        i_index = 0
+
+    ncells = int(f.readline())
+    tmap = netgen_to_meshio_type[dim]
+
+    for i in range(ncells):
+        line, is_eof = _fast_forward_over_blank_lines(f)
+        data = list(filter(None, line.split(" ")))
+        index = int(data[i_index])
+        if dim == 2:
+            nump = int(data[4])
+            pi0 = 5
+        elif dim == 3:
+            nump = int(data[1])
+            pi0 = 2
+
+        pi = list(map(int, data[pi0 : pi0 + nump]))
+
+        t = tmap[nump]
+        if len(cells) == 0 or t != cells[-1][0]:
+            cells.append((t, []))
+            cells_index.append([])
+        cells[-1][1].append(pi)
+        cells_index[-1].append(index)
+
+
+def _write_cells(f, block, index=None):
+    if len(block) == 0:
+        return
+    pmap = np.array(meshio_to_netgen_pmap[block.type])
+    dim = _topological_dimension[block.type]
+    post_data = []
+    pre_data = []
+    i_index = 0
+    if dim == 0:
+        post_data = [1]
+        i_index = 1
+    if dim == 1:
+        pre_data = [1, 0]
+        post_data = [-1, -1, 0, 0, 1, 0, 1, 0]
+    if dim == 2:
+        pre_data = [1, 1, 0, 0, len(pmap)]
+        i_index = 1
+    if dim == 3:
+        pre_data = [1, len(pmap)]
+
+    col1 = len(pre_data)
+    col2 = col1 + len(pmap)
+    col3 = col2 + len(post_data)
+
+    pi = np.zeros((len(block), col3), dtype=np.int32)
+    pi[:, :col1] = np.repeat([pre_data], len(block), axis=0)
+    pi[:, col1:col2] = block.data[:, pmap] + 1
+    pi[:, col2:] = np.repeat([post_data], len(block), axis=0)
+    if index is not None:
+        pi[:, i_index] = index
+    np.savetxt(f, pi, "%i")
+
+
+def _skip_block(f):
+    n = int(f.readline())
+    for i in range(n):
+        f.readline()
+
+
+def read_buffer(f):
+    points = []
+    cells = []
+    cells_index = []
+
+    while True:
+        # fast-forward over blank lines
+        line, is_eof = _fast_forward_over_blank_lines(f)
+        if is_eof:
+            break
+
+        line = line.strip()
+        if line.startswith("#"):
+            continue
+        elif line == "dimension":
+            dimension = int(f.readline())
+        elif line == "geomtype":
+            f.readline()
+        elif line == "mesh3d":
+            continue
+        elif line == "points":
+            npoints = int(f.readline())
+            for i in range(npoints):
```

Comment:
We can merge this, but all these loops are super slow. There are faster alternatives, you may want to check out other ascii-like file types in the future.